### ISSUE TITLE
Rule_LT08 does not account for comma style variations

### ISSUE THREAD

**Issue by jcrada** [sqlfluff/sqlfluff#7775]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When sqlfluff is configured with 
```
[sqlfluff:layout:type:comma]
line_position = leading:align-following
```
and `RuleLT_08` applies
then I get `NotImplementedError` because `global_comma_style` checks only for "trailing" or "leading"

Reference:
- Rule_LT08: [github](https://github.com/sqlfluff/sqlfluff/blob/ed49defe6f5411c4fc9d7db1c5c51d7d3a99b7f8/src/sqlfluff/rules/layout/LT08.py#L171)

### Expected Behaviour

`leading:align-following` should be an [acceptable variation](https://docs.sqlfluff.com/en/stable/configuration/layout.html) to the rule.
Perhaps instead of comparing with equality (`global_comma_style="leading"`), you should be comparing with `startswith` (`global_comma_style.startswith("leading")`)

### Observed Behaviour

```
CRITICAL   [LT08] Applying rule LT08 to 'packages/kraken/analytics/territories/gbr/markets/elec/models/facts/inter_electricity_smart_meters_with_accumulation_read_in_last_month_pii.sql' threw an Exception: Unexpected global comma style 'leading:align-following' 
Traceback (most recent call last):
  File "/Users/juan.rada-vilela/Development/data/ktl-dbt-clients-kap/.venv/lib/python3.13/site-packages/sqlfluff/core/rules/base.py", line 518, in crawl
    res = self._eval(context=context)
  File "/Users/juan.rada-vilela/Development/data/ktl-dbt-clients-kap/.venv/lib/python3.13/site-packages/sqlfluff/rules/layout/LT08.py", line 161, in _eval
    raise NotImplementedError(
        f"Unexpected global comma style {global_comma_style!r}"
    )
NotImplementedError: Unexpected global comma style 'leading:align-following'
```

### How to reproduce

```
WITH
    smart_meter AS (
        SELECT TRUE AS is_true

    ), recent_smart_meter_reads AS (
        SELECT
            is_true
          , FALSE AS is_false
        FROM smart_meter
    )
SELECT * FROM recent_smart_meter_reads
```

### Dialect

databricks

### Version

sqlfluff, version 4.0.4

### Configuration

```
[sqlfluff]
dialect = databricks
templater = jinja
indent_unit = space
rules = aliasing,ambiguous,capitalisation,convention,jinja,layout,references,structure
exclude_rules=layout.long_lines,references.keywords,structure.column_order

[sqlfluff:indentation]
tab_space_size = 4
indented_joins = true
indented_ctes = true

[sqlfluff:layout:type:comma]
line_position = leading:align-following
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @jcrada , this exception throwing has been fixed recently in #7702.

It has not yet been released, but locally I have tested the above and I don't see any exception, but I see the following output:
```bash
sqlfluff fix test.sql 
==== finding fixable violations ====
WARNING    One fix for LT01 not applied, it would re-cause the same error.                                                   
WARNING    One fix for LT01 not applied, it would re-cause the same error.                                                   
WARNING    One fix for LT01 not applied, it would re-cause the same error.                                                   
WARNING    One fix for LT01 not applied, it would re-cause the same error.                                                   
WARNING    One fix for LT01 not applied, it would re-cause the same error.                                                   
WARNING    One fix for LT01 not applied, it would re-cause the same error.                                                   
WARNING    One fix for LT01 not applied, it would re-cause the same error.                                                   
WARNING    One fix for LT01 not applied, it would re-cause the same error.                                                   
WARNING    Loop limit on fixes reached [10].                                                                                 
==== no fixable linting violations found ====
All Finished 📜 🎉!
```

That LT01 is complaining due to line 5: the comma before the second CTE not being on a new line. If you change that, it can properly fix:
```bash
sqlfluff fix test.sql -d databricks
==== finding fixable violations ====
== [test.sql] FAIL                                                                                                           
L:   5 | P:   6 | LT04 | Found trailing comma ','. Expected only leading near
                       | line breaks. [layout.commas]
L:   6 | P:   5 | LT08 | Blank line expected but not found after CTE closing
                       | bracket. [layout.cte_newline]
L:  12 | P:   1 | LT08 | Blank line expected but not found after CTE closing
                       | bracket. [layout.cte_newline]
L:  12 | P:  39 | LT12 | Files must end with a single trailing newline.
                       | [layout.end_of_file]
== [test.sql] FIXED
4 fixable linting violations found
```

And this is the result:
```sql
WITH
    smart_meter AS (
        SELECT TRUE AS is_true

    )

  , recent_smart_meter_reads AS (
        SELECT
            is_true
          , FALSE AS is_false
        FROM smart_meter
    )

SELECT * FROM recent_smart_meter_reads
```

**Comment by jcrada** [comment]
Oh thanks. The error is exactly where you indicate. It has to be manually changed so it can be formatted accordingly.
I missed that merged PR. 
All good then!

Thanks for the great work!
